app/views/shared/_post_likes_preview.html.erb

```diff
@@ -0,0 +1,13 @@
+<div class="flex gap-4 items-center">
+  <!-- Like Button (Like Action) -->
+  <%= form_with(model: [@user, @post, Like.new], url: user_post_likes_path(@user, @post), method: :post, local: true, class: "like-form w-fit text-lg font-bold text-red-700 border-2 rounded-full px-2", data: { turbo_frame: "posts" }) do |form| %>
+    <%= form.submit 'Like', class: "cursor-pointer", disabled: @like.present? %>
+  <% end %>
+
+  <!-- Unlike Button (Unlike Action) -->
+  <% if @like.present? %>
+    <%= form_with(model: [@user, @post, @like], url: user_post_like_path(@user, @post, @like), method: :delete, local: true, class: "like-form w-fit text-lg font-bold text-slate-500 border-2 rounded-full px-2", data: { turbo_frame: "posts" }) do |form| %>
+      <%= form.submit 'Dislike', class: "cursor-pointer" %>
+    <% end %>
+  <% end %>
```

Comment:
👍 Great job on adding the like and dislike buttons! Here's a suggestion to enhance the user experience:

- [ ] [optional] Currently, you're hiding the dislike button initially, but after liking a post, both like and dislike buttons are visible. To enhance the user interface and maintain consistency, consider hiding the like button again after the user clicks the like button. This way, the user won't see both buttons simultaneously.

https://github.com/lily4178993/nell-blog/assets/55840999/11b2a485-2efe-4702-9ed6-6d3ecb0166d6

This adjustment will improve the visual presentation and ensure a cleaner user interface. Keep up the good work! 🚀👍

Reply:
Thank you for highlighting the like/dislike button issue. I've revised the code to ensure a seamless transition between the buttons. Your feedback is valuable. You can see the changes [here](https://github.com/lily4178993/nell-blog/pull/7/commits/79d559a10f2c4d9bb170cfdffe83253af8225d88).